(AEAP) Revert #3491, instead only replace contacts in verified groups

Hocuri opened the pull request:
#3491 introduced a bug that your address is only replaced in the first group you write to, which was rather hard to fix. In order to be able to release something, we agreed to revert it and instead only replace the contacts in verified groups (and in broadcast lists, if the signing key is verified).

More ideas for later:

- dkim verification to generally make from-forgery harder (and we could use it as a neccessary signal for autocrypt key changes) (see #3507 and #3508)
- making the AEAP change into an interactive UX: if you change your address and post to an opportunistic group, someone will have to say "ok, perform the transition" and will then send an added/removed message. this requires UI changes/additions.
- tracking when autocrypt keys changed and how often a key was used -- if it's not used enough AEAP-transition will be rejected on the receiving side.

(partly quoting @hpk42 here)

Depends on #3505.

hpk42 commented on src/peerstate.rs:
```diff
@@ -596,63 +635,15 @@ pub async fn maybe_do_aeap_transition(
-                    .query_get_value(
-                        "SELECT id FROM contacts WHERE addr=? COLLATE NOCASE;",
-                        paramsv![peerstate.addr],
+                // Add an info messages to all chats with this contact
```
```suggestion
                // Add an info messages to all chats with this (verified) contact
```

Hocuri replied:
```suggestion
                // Add info messages to chats with this (verified) contact
```

hpk42 reviewed (approved):
lgtm, just a minor nit. also nice it's not a big PR (mostly a revert anyway).

Hocuri commented:
Ah, yes, rebasing after squash merging is hard.

I rebased again using `git rebase c57d98f --onto origin/master -X theirs`, removed some lines that had somehow sneaked in (82eb089b), checked that the code was now the same as before the rebase using `git diff origin/hoc/aeap-dev-msg-and-only-in-verified-group` (my branch was the same as origin before the rebase), and then force pushed (`git push --force-with-lease`)
